### DISCUSSION TITLE
How to implement assertions with a tolerance

### DISCUSSION THREAD

**Discussion by rassie** [syrupy-project/syrupy#537] [Q&A]
I would like to implement this assertion to check that `count` is within 10% of the expected value:

```python
assert count <= snapshot * 1.1 and count >= snapshot * 0.9
```

Is there a good way to do that with `syrupy`?

**Comment by noahnu** [comment]
Sorry about the delayed response, I didn't even realize we had discussions enabled :sweat:. We have notifications setup for new issues, but nothing watching the discussions.

---

I believe the author of https://github.com/tophat/syrupy/issues/438 was trying to do the same sort of assertion as you describe just using pytest.approx instead. There are some suggestions and workarounds in that issue. Please let me know if there are any follow up questions.

In terms of that specific syntax, it's not possible with the latest syrupy, though it's interesting and may warrant further investigation. What would the underlying snapshot look like?

**Reply by rassie** [reply]
I would assume a snapshot would just contain a number, which would be the 100% baseline. When running the test, it should match with a provided tolerance. I don't really know what would be a good syntax for that, but I suppose anything will work as long it does what's needed.

**Reply by noahnu** [reply]
> it should match with a provided tolerance

The difficulty is that we only serialize data. We don't currently support deserialization. So we do a character by character match on the serialized data. This ensures the snapshots are 100% deterministic (unlike snapshottest which writes python that can include memory addresses for example and thus produce inconsistent results between snapshots).
